packages/webpack-plugin/index.js

```diff
@@ -0,0 +1,150 @@
+          name: 'PrismaPlugin',
+          stage: Compilation.PROCESS_ASSETS_STAGE_ANALYSE,
+        },
+        async (assets) => {
+          const jsAssetNames = Object.keys(assets).filter((k) => k.endsWith('.js'))
+          const jsAsyncActions = jsAssetNames.map(async (assetName) => {
+            // prepare paths
+            const outputDir = compiler.outputPath
+            const assetPath = path.resolve(outputDir, assetName)
+            const assetDir = path.dirname(assetPath)
+
+            // get sources
+            const oldSourceAsset = compilation.getAsset(assetName)
+            const oldSourceContents = oldSourceAsset.source.source() + ''
+
+            // update sources
+            for (const match of oldSourceContents.matchAll(prismaDirRegex)) {
+              const prismaDir = await getPrismaDir(match[1])
+              const prismaFiles = await getPrismaFiles(match[1])
+
+              prismaFiles.forEach((f) => {
+                const from = path.join(prismaDir, f)
+
+                // if we have multiple schema.prisma files, we need to rename them
+                if (f === 'schema.prisma' && fromDestPrismaMap[from] === undefined) {
+                  f += ++schemaCount
+                }
+
+                // if we already have renamed it, we need to get its "renamed" name
+                if (f.includes('schema.prisma') && fromDestPrismaMap[from] !== undefined) {
+                  f = path.basename(fromDestPrismaMap[from])
+                }
+
+                if (f.includes('schema.prisma')) {
+                  // update "schema.prisma" to "schema.prisma{number}" in the sources
+                  const newSourceString = oldSourceContents.replace(/schema\.prisma/g, f)
+                  const newRawSource = new sources.RawSource(newSourceString)
+                  compilation.updateAsset(assetName, newRawSource)
+                }
+
+                // update copy map
+                fromDestPrismaMap[from] = path.join(assetDir, f)
+              })
+            }
+          })
+
+          await Promise.all(jsAsyncActions)
+        },
+      )
+    })
+
+    // update nft.json files to include prisma files (only for next.js)
+    compiler.hooks.compilation.tap('PrismaPlugin', (compilation) => {
+      compilation.hooks.processAssets.tapPromise(
+        {
+          name: 'PrismaPlugin',
+          stage: Compilation.PROCESS_ASSETS_STAGE_ANALYSE,
+        },
+        async (assets) => {
+          const nftAssetNames = Object.keys(assets).filter((k) => k.endsWith('.nft.json'))
+          const nftAsyncActions = nftAssetNames.map((assetName) => {
+            // prepare paths
+            const outputDir = compiler.outputPath
+            const assetPath = path.resolve(outputDir, assetName)
+            const assetDir = path.dirname(assetPath)
+
+            // get sources
+            const oldSourceAsset = compilation.getAsset(assetName)
+            const oldSourceContents = oldSourceAsset.source.source() + ''
```

Comment:
is `+ ''` here to coerce the return value of `oldSourceAsset.source.source()` to string?

Reply:
It's a small operation to make the type-checking work (here in JS) otherwise it could be undefined. That said it can never be undefined because we are directly using the `assetName` given by webpack.